### ISSUE TITLE
When placeholder is not '', the iframe body does not has correct role attribute.

### ISSUE THREAD

**Issue by BetaZhang** [tinymce/tinymce#9231]
## 📝 Provide detailed reproduction steps (if any)
Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](fiddle.tiny.cloud) or similar.
1. Set editor placeholder to 'Enter the question content'
2. Use axe DevTools scan the editor.

### ✔️ Expected result
*What is the expected result of the above steps?*
The iframe body should have role='textbox'.

### ❌ Actual result
*What is the actual result of the above steps?*
The iframe does not have role.
![WXWorkCapture_17019991438659](https://github.com/tinymce/tinymce/assets/102902468/c9946225-18f7-4c8e-b5d2-ad1e006fdd02)

### ❓ Possible solution
*If you have ideas, you can list them here. Otherwise, you can delete this section.*
Add role='textbox' to iframe body.

## 📃 Other details
- Browser: Chrmoe
- OS: Windows
- First affected version: …
- Worked in version: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by TheSpyder** [comment]
Interesting. We added the `aria-placeholder` because we wanted to give screen readers some way to know that the content in the editor was a placeholder. I don't think we realised this required the `textbox` role on the content.

Textbox doesn't seem like an appropriate role for a rich text editor, so our only option is to remove the attribute. That seems like it will make our placeholders a poor experience for screen readers though.
